### PR TITLE
HLint and stylish-haskell suggestions

### PR DESCRIPTION
**Pull request by varosi**
Those suggestions were extracted from https://github.com/snoyberg/http-client/pull/274 review.

### CODE REVIEW

File: http-client/Network/HTTP/Client/Manager.hs [hunk 1]
```diff
@@ -94,15 +99,15 @@ defaultManagerSettings = ManagerSettings
         case fromException e of
             Just (_ :: IOException) -> True
             _ ->
-                case fmap unHttpExceptionContentWrapper $ fromException e of
+                case unHttpExceptionContentWrapper <$> fromException e of
```
Comment (snoyberg) [inline http-client/Network/HTTP/Client/Manager.hs]: It looks like this change will introduce warnings in newer GHCs.

File: http-client/Network/HTTP/Client/Manager.hs [hunk 2]
```diff
@@ -94,15 +99,15 @@ defaultManagerSettings = ManagerSettings
...
                     -- Note: Some servers will timeout connections by accepting
                     -- the incoming packets for the new request, but closing
                     -- the connection as soon as we try to read. To make sure
                     -- we open a new connection under these circumstances, we
                     -- check for the NoResponseDataReceived exception.
                     Just NoResponseDataReceived -> True
-                    Just IncompleteHeaders -> True
-                    _ -> False
+                    Just IncompleteHeaders      -> True
```
Comment (snoyberg) [inline http-client/Network/HTTP/Client/Manager.hs]: I'm -1 on such alignment changes; it encourages noisy diffs for future changes.

File: http-client/Network/HTTP/Client/Manager.hs [hunk 3]
```diff
@@ -212,9 +217,7 @@ reap baton wmapRef =
     loop = do
         threadDelay (5 * 1000 * 1000)
         mmapRef <- deRefWeak wmapRef
-        case mmapRef of
```
Comment (snoyberg) [inline http-client/Network/HTTP/Client/Manager.hs]: I believe this form is less clear, and in particular the comment explaining what the `Nothing` case means is lost.

File: http-client/Network/HTTP/Client/Manager.hs (hunk 2 above)
Comment (snoyberg) [inline http-client/Network/HTTP/Client/Manager.hs]: Note that this applies to many other diffs in this PR.

**Comment by varosi** [comment]
I think that all the changes are not much useful, so I'm closing it.

**Pull request closed without merging**